Cython/Utility/CommonStructures.c

```diff
@@ -7,98 +7,90 @@ static PyObject* __Pyx_FetchCommonTypeFromSpec(PyType_Spec *spec, PyObject *base
 
 /////////////// FetchCommonType ///////////////
 
-static PyTypeObject* __Pyx_FetchCommonType(PyTypeObject* type) {
-    PyObject* fake_module;
-    PyTypeObject* cached_type = NULL;
-
-    fake_module = PyImport_AddModule((char*) "_cython_" CYTHON_ABI);
+static PyObject *fetch_fake_module(void) {
+    PyObject *fake_module = PyImport_AddModule((char*) "_cython_" CYTHON_ABI);
     if (!fake_module) return NULL;
     Py_INCREF(fake_module);
+    return fake_module;
+}
+
+static void verify_cached_type(PyObject *cached_type,
+                               const char *name,
+                               Py_ssize_t basicsize,
+                               Py_ssize_t expected_basicsize) {
+    if (!PyType_Check(cached_type)) {
+        PyErr_Format(PyExc_TypeError,
+            "Shared Cython type %.200s is not a type object", name);
+    }
+    if (basicsize != expected_basicsize) {
+        PyErr_Format(PyExc_TypeError,
+            "Shared Cython type %.200s has the wrong size, try recompiling",
+            name);
+    }
+}
+
+static PyTypeObject* __Pyx_FetchCommonType(PyTypeObject* type) {
+    PyObject* fake_module;
+    PyTypeObject *cached_type = NULL;
 
+    fake_module = fetch_fake_module();
+    if (!fake_module) goto done;
```

Comment:
The common pattern in the code base is to `goto bad` on failures, so that error cleanup can happen. It's often even used if the error case is no different from the normal function exit, because it's such a future proof and nicely declarative pattern. However, since `cached_type` can be set to non-NULL also in error cases in this function, it also seems necessary here.

Reply:
I opted to just re-use what was already there which works just fine!

Reply:
Done